DataDog/datadog-operator
# Add beta support for Agent Data Plane.

### What does this PR do?

Adds support for the experimental Agent Data Plane project. This follows a similar path to #1269 where only an experimental CLI flag is exposed for enabling ADP support, and we're leaving the CRDs alone.

### Motivation

The ADP deployment model is based around a new container, so Operator changes are necessary for internal testing.

### Additional Notes

N/A

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

Not tied to a standard Agent version, as this depends on a custom internal-only Datadog Agent build.

### Describe your test plan

- Deploy the custom Operator image to a designated staging cluster, and enable the newly-added CLI flag.
- Update that cluster to use an ADP-specific image for the Datadog Agent.
- Observe that the new container for ADP is created.

### Checklist
